### DISCUSSION TITLE
Pipe Error 32?

### DISCUSSION THREAD

**Discussion by ssokol** [VCTLabs/cp210x-program#30] [Q&A]
I am getting a pipe error (32) when I execute cp210x-program. According to both the manufacturer and lsusb, the device is a CP2102. cp210x-program works perfectly with other adapters.

```
ssokol@fbintel:~/cp210x-program$ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 8087:0a2a Intel Corp.
Bus 001 Device 005: ID 10c4:ea60 Cygnal Integrated Products, Inc. CP2102/CP2109 UART Bridge Controller [CP210x family]
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
ssokol@fbintel:~/cp210x-program$ sudo ./cp210x-program
[Errno 32] Pipe error
ssokol@fbintel:~/cp210x-program$
```

Any suggestions or thoughts?

**Comment by konasimakopoulos** [comment]
Just my 2 cents: this seems to happen with a CP2102N but not with the older CP2102.

**Reply by sarnold** [reply]
In spite of the product string above, that part is a cp2109 and *not* the legacy cp2102.  I updated the readme doc links and part descriptions/status, and the current (legacy) code will most likely *only* work with (legacy) cp2102/cp2103, and possibly cp2101 parts.

**Comment by Roland-aixACCT-Systems** [comment]
Dear all,
we are considering using the "cp210x-program" as a manufacturing tool for the CP2102N.  I checked the program with a demo board and I get exactly the same error.
Can you confirm that the code is not compatible with the CP2102N?
Thank you in advance for your answer!
Best regards,
Roland

**Reply by sarnold** [reply]
Correct, not compatible with cp2102N (which has a totally different EEPROM layout/interface).

**Reply by Roland-aixACCT-Systems** [reply]
Thank you very much for confirming this.
Best regards,
Roland

**Reply by sarnold** [reply]
There's a new(ish) N-compatible tool you might want to look at (recently merged to master). @Roland-aixACCT-Systems 
see the minimal usage demo in https://github.com/VCTLabs/cp210x-program/discussions/33

**Comment by sarnold** [comment]
We would need some "N" hardware to test, but for now I'd consider this a valid bug (given the information above).

**Comment by sarnold** [comment]
Same problem with CP2105 (after unloading the kernel module):

```
Bus 001 Device 068: ID 10c4:ea70 Silicon Labs CP2105 Dual UART Bridge

 $ ./cp210x-program --read-cp210x -m 10c4:ea70
[Errno 32] Pipe error
```

**Comment by PeterBorisenko** [comment]
Getting same error with CP2104
```
$ python ./cp210x-program --read-cp210x
Cp210xProgrammer returned obj:
<cp210x.cp210x.Cp210xProgrammer object at 0x7fdb73d7b520>
Cp210xProgrammer instance is valid!!
usb ctrl msg: request, value, index, length:
  255 14089 0 1024
[Errno 32] Pipe error
```
